app/controllers/application_controller.rb

```diff
@@ -1,3 +1,11 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  rescue_from (Errno::ECONNREFUSED), with: :record_not_found
+  rescue_from ActiveRecord::RecordNotFound, with: :record_not_found
+  rescue_from StandardError, with: :record_not_found
```

Comment:
`Errno::ECONNREFUSED` and `StandardError` should probably not be resulting in a 404 — and definitely should not be swallowed quietly by the handler.